hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ATemporaryCredentials.java

```diff
@@ -163,6 +163,37 @@ public void testSTS() throws IOException {
     }
   }
 
+  /**
+   * Test use of Invalid STS for requesting temporary credentials.
+   *
+   * The property test.sts.endpoint can be set to point this at different
+   * STS endpoints. This test will use the AWS credentials (if provided) for
+   * S3A tests to request temporary credentials, then attempt to use those
+   * credentials instead.
+   *
+   * @throws IOException failure
+   */
+  @Test
+  public void testSTSInvalid() throws IOException {
+    Configuration conf = getContract().getConf();
+    S3AFileSystem testFS = getFileSystem();
+    credentials = getS3AInternals().shareCredentials("testSTS");
+
+    String bucket = testFS.getBucket();
+    try {
```

Comment:
Look at the other tests in this class, `testSessionCredentialsRegionBadEndpoint()` for example. We can write this similarly. 

You will also need to add a test case to check if China endpoint works.

Reply:
Done, I have added test for most of the possible region combinations including china endpoints. Please do check.